pkg/services/provisioning/apps/config_reader.go

```diff
@@ -0,0 +1,130 @@
+package apps
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/plugins"
+	"gopkg.in/yaml.v2"
+)
+
+type configReader interface {
+	readConfig(path string) ([]*appsAsConfig, error)
+}
+
+type configReaderImpl struct {
+	log log.Logger
+}
+
+func newConfigReader(logger log.Logger) configReader {
+	return &configReaderImpl{log: logger}
+}
+
+func (cr *configReaderImpl) readConfig(path string) ([]*appsAsConfig, error) {
+	var apps []*appsAsConfig
+	cr.log.Debug("Looking for app provisioning files", "path", path)
+
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		cr.log.Error("Failed to read app provisioning files from directory", "path", path, "error", err)
```

Comment:
Should this be a warning instead, since we don't treat it as an error?

Reply:
We use error level for data source and notifiers so something to consider for later.